source/how-tos/enable-intel-tsn.rst

```diff
@@ -0,0 +1,280 @@
+.. _tsn:
+
+TSN
+~~~~~~~~~~~~~~~~~
+
+Overview
+--------------------
+
+TSN (Time Sensitive Networking) solution is a set of capabilities that offers Time Synchronization, Traffic Shaping, set of IEEE open standard communication by bringing industrial-grade robustness and reliability to Ethernet . TSN removes the need for physical separation of critical and non-critical communication networks, thus allowing open data exchange between operations and enterprise - a concept at the heart of the Industrial Internel of Things (IIoT).
+To learn more about TSN, see |TSN| for details.
+
+.. |TSN| raw:: html
+
+   <a href="https://www.intel.com/content/www/us/en/industrial-automation/programmable/applications/automation/tsn.html" target="_blank">TSN</a>
+
+
+**TSN MAC Address**
+
+User needs to provide TSN MAC address that is unique per board and that can be independently updateable. FBU tool is used to create a raw TSN MAC binary. Details on how this is packaged and processed will
+be discussed in the later sections.
+
+SBL defines one TSN container inside its BIOS region. This container has the TSN MAC Address binary that can be updated via SBL firmware update mechanism. During boot time, SBL would load TSN MAC binary from the TSN container and process it at Stage 2 during Runtime.
+
+User could generate the raw TSN MAC Address binary with the help of FBU (Firmware and BIOS Utility) Tools. This raw binary need to be signed using the same key used in SBL build so that it could pass SBL verification during boot time.
+Now the signed TSN is kept into SBL container and encapsulate it to create a capsule image which is then signed by a firmware update key. The firmware update key should match with SBL key so that the capsule image could pass SBL capsule verification during firmware update.
+The capsule location could be specified by SBL configuration data. SBL will update the TSN container using the specified capsule image during the next boot once firmware update is triggered.
+
+
+TSN enable
+--------------------
+
+For each platform, there is a build configuration file BoardConfig.py. User could check this build configuration  file to make sure self.ENABLE_TSN_MAC_ADDRESS is set to 1 to enable TSN MAC Address binary.
+The other TSN related configuration data is PchTsnEnable. Make sure to set this to 1 under Silicon Configuration Data.
+
+.. _tsn_container:
+
+TSN container
+--------------------
+
+Container support is a generic SBL feature. Please refer to |Container| for generic details.
+
+.. |Container| raw:: html
+
+   <a href="https://slimbootloader.github.io/security/container-security.html?highlight=container#container-security" target="_blank">Container</a>
+
+
+SBL defines one such container for TSN binary. The raw TSN MAC binary is packaged as a container in SBL.
+SBL loads this container during Runtime and is processed by Intel® Firmware Support Package (Intel® FSP) during Runtime.

```

Comment:
SBL loads this container during Runtime and processes it as required including passing the relevant settings to Intel® Firmware Support Package